pallets/parachain-staking/src/types.rs

```diff
@@ -175,6 +175,7 @@ where
 		}
 	}
 
+	#[allow(clippy::indexing_slicing)]
```

Comment:
We should not allow index slicing, but rather use `expect`, so it is made explicit where we are making assumptions.